### PR TITLE
povray: Bump to povray 3.7.0.8

### LINKED ISSUE #3297: Povray crashes on x86_64
Attaching debug report

### PR DESCRIPTION
**Pull request by kenmays**
**POV-Ray - The Persistence of Vision Ray Tracer**
The  Persistence  of  Vision(tm)   Ray-Tracer   creates   three-dimensional,
photo-realistic images using a rendering  technique  called  ray-tracing.  It
reads in a text  file  containing  information  describing  the  objects  and
lighting in a scene and generates an image of that scene from the view  point
of a camera also described in the  text  file.  Ray-tracing  is  not  a  fast
process by any means, but it produces very high quality images with realistic
reflections, shading, perspective and other effects.

Hall of Fame renderings: http://hof.povray.org/ 
Docs: http://www.povray.org/documentation/
URL: http://www.povray.org/

NOTE:
- Built successfully and passed 'make check' tests on hrev53121 x86/GCC 7.3.0.

Closes #3297

### CODE REVIEW

**Comment by kenmays** [comment]
```
~/povray> bin/povray --version
POV-Ray 3.7.0.7.unofficial

This is an unofficial version compiled by:
 HaikuPorts
 The POV-Ray Team is not responsible for supporting this version.

Copyright 1991-2013 Persistence of Vision Raytracer Pty. Ltd.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Built-in features:
  I/O restrictions:          enabled
  X Window display:          disabled
  Supported image formats:   gif tga iff ppm pgm hdr png jpeg tiff
  Unsupported image formats: openexr

Compilation settings:
  Build architecture:  i586-pc-haiku
  Built/Optimized for: i586-pc-haiku (using -march=native)
  Compiler vendor:     gnu
  Compiler version:    g++ 7.3.0
  Compiler flags:      -pipe -Wno-multichar -Wno-write-strings -fno-enforce-eh-specs -Wno-non-template-friend -s -O3 -ffast-math -malign-double -march=native -std=c++11 -DBOOST_NO_CXX11_CONSTEXPR -D_BSD_SOURCE
~/povray> uname -a
Haiku shredder 1 hrev53135 May 16 2019 12:22:30 BePC x86 Haiku
~/povray> 
```

**Comment by alaviss** [comment]
>       Built/Optimized for: i586-pc-haiku (using -march=native)

Can you make sure that `-march=native` is not used? This flag optimize the generated code for the machine that builds the code and may cause the generated binary to be unusable/slow on other machines.

**Comment by kenmays** [comment]
alaviss - Patches welcome.

**Comment by kenmays** [comment]
@alaviss - Fixed.

**Comment by kenmays** [comment]
@alaviss - Fixed. No major issues foreseen.

    povray --benchmark = successfully completed, 1 hour, 1 min on 1.73Ghz Intel Core i3 on Haiku 32-bit
        Even @**128 threads** setting - no crashes on Haiku.
        Compared to 400Mhz Intel Pentium II, 4 hours 19 minutes

**Comment by kenmays** [comment]
No issues.
```
~/povray> bin/povray --version
POV-Ray 3.7.0.8.unofficial

This is an unofficial version compiled by:
 HaikuPorts
 The POV-Ray Team is not responsible for supporting this version.

Copyright 1991-2013 Persistence of Vision Raytracer Pty. Ltd.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Built-in features:
  I/O restrictions:          enabled
  X Window display:          disabled
  Supported image formats:   gif tga iff ppm pgm hdr png jpeg tiff
  Unsupported image formats: openexr

Compilation settings:
  Build architecture:  i586-pc-haiku
  Built/Optimized for: i586-pc-haiku (using -march=native)
  Compiler vendor:     gnu
  Compiler version:    g++ 7.3.0
  Compiler flags:      -pipe -Wno-multichar -Wno-write-strings -fno-enforce-eh-specs -Wno-non-template-friend -s -O3 -ffast-math -malign-double -march=native -march=pentium-m -mtune=generic -std=c++11 -DBOOST_NO_CXX11_CONSTEXPR -D_BSD_SOURCE
~/povray> uname -a
Haiku shredder 1 hrev53144 May 18 2019 11:21:28 BePC x86 Haiku
~/povray>
```

**Comment by kenmays** [comment]
No issues.

~/povray> bin/povray --version
POV-Ray 3.7.0.8.unofficial

This is an unofficial version compiled by:
 HaikuPorts
 The POV-Ray Team is not responsible for supporting this version.

Copyright 1991-2013 Persistence of Vision Raytracer Pty. Ltd.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Built-in features:
  I/O restrictions:          enabled
  X Window display:          disabled
  Supported image formats:   gif tga iff ppm pgm hdr png jpeg tiff
  Unsupported image formats: openexr

Compilation settings:
  Build architecture:  i586-pc-haiku
  Built/Optimized for: i586-pc-haiku (using -march=native)
  Compiler vendor:     gnu
  Compiler version:    g++ 7.3.0
  Compiler flags:      -pipe -Wno-multichar -Wno-write-strings -fno-enforce-eh-specs -Wno-non-template-friend -s -O3 -ffast-math -malign-double -march=native -march=pentium -mtune=generic -std=c++11 -DBOOST_NO_CXX11_CONSTEXPR -D_BSD_SOURCE
~/povray> uname -a
Haiku shredder 1 hrev53144 May 18 2019 11:21:28 BePC x86 Haiku
~/povray> povray -benchmark
**Trace Time: 1 hours, 2 minutes, 9 seconds (3729.823 seconds) using 8 threads**

File: media-gfx/povray/povray-3.7.0.8.recipe [hunk 1]
```diff
@@ -77,14 +75,24 @@ BUILD()
...
+			COMPILED_BY="HaikuPorts" \
+			--without-openexr \
+			--with-boost-libdir=/boot/system/lib/x86
+	fi
+	if [ $effectiveTargetArchitecture = x86_64 ]; then
+		export CXXFLAGS="-march=core2 -mtune=generic -std=c++11 -DBOOST_NO_CXX11_CONSTEXPR -D_BSD_SOURCE"
+		runConfigure ./configure \
+			COMPILED_BY="HaikuPorts" \
+			--with-boost-libdir=`finddir B_SYSTEM_LIB_DIRECTORY`
+	fi
```
Comment (alaviss) [inline media-gfx/povray/povray-3.7.0.8.recipe]: Instead of this, simply adding the option `--disable-optimiz` should take care of `-march=native`

Comment (kenmays) [reply media-gfx/povray/povray-3.7.0.8.recipe]: Fixed.

File: media-gfx/povray/povray-3.7.0.8.recipe [hunk 2]
```diff
@@ -58,9 +58,7 @@ BUILD_REQUIRES="
 	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	cmd:aclocal
 	cmd:autoconf
-	cmd:autoheader
 	cmd:automake
 	cmd:awk
 	cmd:find
```
Comment (korli) [inline media-gfx/povray/povray-3.7.0.8.recipe]: I would add cmd:autoupdate

Comment (kenmays) [reply media-gfx/povray/povray-3.7.0.8.recipe]: Fixed.

File: media-gfx/povray/povray-3.7.0.8.recipe [hunk 3]
```diff
@@ -77,14 +75,25 @@ BUILD()
...
 	cd ..
 
-	export CXXFLAGS="-std=c++11 -DBOOST_NO_CXX11_CONSTEXPR -D_BSD_SOURCE"
 	export LDFLAGS="-lbsd -lnetwork"
+	export CXXFLAGS="-std=c++11 -DBOOST_NO_CXX11_CONSTEXPR -D_BSD_SOURCE"
 
-	runConfigure ./configure \
-		COMPILED_BY="HaikuPorts" \
-		--with-boost-libdir=`finddir B_SYSTEM_LIB_DIRECTORY`
+	if [ $effectiveTargetArchitecture = x86_gcc2 ]; then
```
Comment (korli) [inline media-gfx/povray/povray-3.7.0.8.recipe]: Please use a configureArgs variable with additional architecture parameters. runConfigure should appear one time.

Comment (kenmays) [reply media-gfx/povray/povray-3.7.0.8.recipe]: Fixed.

Comment (korli) [reply media-gfx/povray/povray-3.7.0.8.recipe]: The fix is not a fix.

Comment (kenmays) [reply media-gfx/povray/povray-3.7.0.8.recipe]: I simplified it. If you have an official solution, patches welcome...

File: media-gfx/povray/povray-3.7.0.8.recipe [hunk 4]
```diff
@@ -82,16 +82,17 @@ BUILD()
 	export LDFLAGS="-lbsd -lnetwork"
 	export CXXFLAGS="-std=c++11 -DBOOST_NO_CXX11_CONSTEXPR -D_BSD_SOURCE"
 
-	runConfigure ./configure \
+if [ $effectiveTargetArchitecture = x86_gcc2 ]; then
+	BoostLibLctr=/boot/system/lib/x86
```
Comment (korli) [inline media-gfx/povray/povray-3.7.0.8.recipe]: is it different from $libDir?

File: media-gfx/povray/povray-3.7.0.8.recipe [hunk 5]
```diff
@@ -82,16 +82,17 @@ BUILD()
...
+else
+	BoostLibLctr=`finddir B_SYSTEM_LIB_DIRECTORY`
+fi
+
+runConfigure ./configure \
```
Comment (korli) [inline media-gfx/povray/povray-3.7.0.8.recipe]: This and the if else fi block needs a tab.

File: media-gfx/povray/povray-3.7.0.8.recipe (hunk 4 above)
Comment (kenmays) [reply media-gfx/povray/povray-3.7.0.8.recipe]: Update recipe.

File: media-gfx/povray/povray-3.7.0.8.recipe (hunk 5 above)
Comment (kenmays) [reply media-gfx/povray/povray-3.7.0.8.recipe]: Looks OK compared to other recipes.

**Pull request closed without merging**